hal/i2c_api.h

```diff
@@ -70,9 +70,72 @@ extern "C" {
 
 /**
  * \defgroup hal_GeneralI2C I2C Configuration Functions
+ *
```

Comment:
It may be good to re-use the relevant existing I2C API defined and undefined behavior from the feature branch here:
https://github.com/ARMmbed/mbed-os/blob/feature-i2c/hal/i2c_api.h#L36-L93